src/components/Input/Input.tsx

```diff
@@ -163,31 +158,14 @@ class Input extends AutoControlledComponent<any, any> {
   }
 
   renderComponent({ ElementType, classes, rest, styles }) {
-    const { children, clearable, input, type } = this.props
+    const { type } = this.props
     const [htmlInputProps, restProps] = this.partitionProps()
 
     const inputClasses = classes.input
-    const iconClasses = classes.icon
-
-    // Render with children
-    // ----------------------------------------
-    if (childrenExist(children)) {
```

Comment:
This change means this component no longer supports children.  Though that might be an acceptable change (not sure of the reasoning behind it) the propTypes are still claiming that this component supports children.

Let's ensure our component features, examples, and propTypes are all consistent.

Reply:
There was a very long conversation about the SUIR Input API which explains why there are children: 
https://github.com/Semantic-Org/Semantic-UI-React/pull/281#issuecomment-248201953

Specifically, children were introduced to support these two cases:
![image](https://user-images.githubusercontent.com/5067638/44606314-88b5f700-a7a1-11e8-97b4-aa18ec706438.png)

There isn't a great shorthand API available to support these cases, as detailed in the conversation linked.  Instead, we opted to support a single label and a single action with the shorthand API.  If users needed to add two labels or two actions, they used children.  In that case, the `<input />`  child is where the component places the actual HTML input, allowing you to choose what is on the left or right of the input.

### Immediate Proposal

For now, I would not support children and I would not support two labels nor two actions.  I would just keep it simple and consistent.  We may need to address this API later to accommodate more advanced usages.

### Possible Future Proposal

My latest thinking on this kind of component would say that the Input should not have any of these concerns (labels and buttons).  Instead, it might be accomplished something like:

```jsx 
<Label attached='end' content='$' />
<Input attached='horizontally' />
<Label attached='start' content='.00' />
```
![image](https://user-images.githubusercontent.com/5067638/44606703-d54e0200-a7a2-11e8-96b6-d47af9358543.png)

and

```jsx 
<Input attached='end' />
<Dropdown attached='horizontally' menu={...} value='articles' />
<Button attached='start' content='Search' />
```
![image](https://user-images.githubusercontent.com/5067638/44606696-cf582100-a7a2-11e8-84f6-d97de88a3b5c.png)

Reply:
@levithomason also regarding the above, I am not sure if it is good to discuss here or to open an RFC for it... regarding the naming of the `<Label>` component, I think it has always introduced a lot of confusion when discussing inputs since there is a literal `<label>` tag which behaves differently than the currently named Label component. Not sure what everyone else's feelings are, but I think it would make a lot of sense to find a different name for the Label component. At least as it pertains to Input components.

I do like the other above proposed simplifications to the shorthand API.